boards/nxp/frdm_rw612/Kconfig.defconfig

```diff
@@ -5,8 +5,13 @@
 
 if BOARD_FRDM_RW612
 
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+DT_CHOSEN_Z_FLASH := zephyr,flash
+
 config FLASH_LOAD_SIZE
-	default 0x400000 if !BOOTLOADER_MCUBOOT && !NXP_MONOLITHIC_BT
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION)) \
+	        if (MCUBOOT || BOOTLOADER_MCUBOOT || (BT_NXP && !NXP_MONOLITHIC_BT))
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_FLASH))
```

Comment:
Hmm..,
The config FLASH_LOAD_SIZE should be fully deleted. As all is done automatically in Kconfig.zephyr
Instead, select **USE_DT_CODE_PARTITION** "Link application into /chosen/zephyr,code-partition from devicetree" where it's needed.
When building MCUBoot, MCUBoot itself will select USE_DT_CODE_PARTITION
If USE_DT_CODE_PARTITION is not selected used whole flash.

Reply:
I have a PR here #76610 that turns out to have the same idea I think as what you said here